luaui/Widgets/gui_chat.lua

```diff
@@ -2588,6 +2596,17 @@ function widget:Initialize()
 		fontsizeMult = value
 		widget:ViewResize()
 	end
+	WG['chat'].addChatLine = function(gameFrame, lineType, name, nameText, text, orgLineID, ignore, chatLineID)
+		addChatLine(gameFrame, lineType, name, nameText, text, orgLineID, ignore, chatLineID, true)
+	end
```

Comment:
Why don't processors belong to gui_chat, removing the need for the ghosting above?

> i'd really like to start adding more API to game widgets, as sometimes the only way to now do things is to fully copy the widget and disable the original, but it's dangerous stuff because if custom widget gets deleted while it disabled the original widget, there is no way to get the original widget on again, and the user is stuck with a broken installation

I'm assuming it was this, I guess what is preventing you from modifying the original widget in this case?